### PR TITLE
Feature/cvsb 181

### PR DESCRIPTION
**Pull request by mgherceanu**
Add preparer page and functionality
Update sync mechanism
Get preparers from storage mechanism.
Save preparers in cache

### CODE REVIEW

File: src/providers/global/sync.service.ts [hunk 1]
```diff
@@ -37,8 +36,10 @@ export class SyncService {
   }
 
   startSync(): void {
-    this.loadOrder.push(this.getAtfs());
-    this.loadOrder.push(this.getDefects());
+    const microservicesListArray: Array<string> = ['Atfs', 'Defects', 'TestTypes', 'Preparers'];
```
Comment (danieliordache) [inline src/providers/global/sync.service.ts]: change 'TestTypes' string to match the change in app.enum

File: src/providers/preparer/preparer.service.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+export class PreparerService {
+
+  constructor(private storageService: StorageService) {
+  }
+
+  getPreparersFromStorage(): Observable<PreparersModel[]> {
+    return from(this.storageService.read(STORAGE.PREPARERS))
+  }
+
+  search(array: PreparersModel[], filter: string) {
```
Comment (danieliordache) [inline src/providers/preparer/preparer.service.ts]: add return type

File: src/providers/preparer/preparer.service.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { Observable } from "rxjs";
+import { from } from "rxjs/observable/from";
+import { STORAGE } from "../../app/app.enums";
+import { StorageService } from "../natives/storage.service";
+import { PreparersModel } from "../../models/preparers/preparers.model";
+
+
+@Injectable()
+export class PreparerService {
```
Comment (danieliordache) [inline src/providers/preparer/preparer.service.ts]: which search do you use? the one from the service or the pipe? remove the unused one

File: src/app/app.enums.ts [hunk 4]
```diff
@@ -1,6 +1,12 @@
 export enum STORAGE {
   ATFS = 'atfs',
-  DEFECTS = 'defects'
+  DEFECTS = 'defects',
+  TESTTYPES = 'test-types',
```
[the remark this answers is not in the document]
Comment (mgherceanu) [reply src/app/app.enums.ts]: done

File: src/pages/testing/test-creation/add-preparer/add-preparer.scss [hunk 5]
```diff
@@ -0,0 +1,47 @@
+page-add-preparer {
+  .button__custom-back {
+    margin: -1px 0 0;
+    display: inherit;
+    min-width: 18px;
+    font-size: 3.4rem;
```
[the remark this answers is not in the document]
Comment (mgherceanu) [reply src/pages/testing/test-creation/add-preparer/add-preparer.scss]: done

File: src/pages/testing/test-creation/add-preparer/add-preparer.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    });
+    alert.present();
+  }
+
+  searchList(ev): void {
+    this.searchValue = ev.target.value;
+    this.filteredPreparers = this.preparerService.search(this.preparers, this.searchValue);
+  }
+
+  detectFocus() {
```
[the remark this answers is not in the document]
Comment (mgherceanu) [reply src/pages/testing/test-creation/add-preparer/add-preparer.ts]: done

File: src/pages/testing/test-creation/test-create/test-create.ts [hunk 7]
```diff
@@ -16,6 +16,7 @@ export class TestCreatePage {
 
   constructor(public navCtrl: NavController, public navParams: NavParams, public phoneService: PhoneService) {
     this.testReport = navParams.get('testReport');
+    console.log('din test create page', this.testReport);
```
[the remark this answers is not in the document]
Comment (mgherceanu) [reply src/pages/testing/test-creation/test-create/test-create.ts]: done

File: src/pages/testing/test-creation/vehicle-details/vehicle-details.ts [hunk 8]
```diff
@@ -31,6 +31,11 @@ export class VehicleDetailsPage {
     }
   }
 
+  goToPreparerPage() {
```
[the remark this answers is not in the document]
Comment (mgherceanu) [reply src/pages/testing/test-creation/vehicle-details/vehicle-details.ts]: done

File: src/providers/preparer/preparer.service.ts (hunk 2 above)
Comment (mgherceanu) [reply src/providers/preparer/preparer.service.ts]: done

File: src/providers/preparer/preparer.service.ts (hunk 3 above)
Comment (mgherceanu) [reply src/providers/preparer/preparer.service.ts]: i removed the pipe, the logic for the search is now in the preparer.service.ts

File: src/providers/global/sync.service.ts (hunk 1 above)
Comment (mgherceanu) [reply src/providers/global/sync.service.ts]: con not modify as it is implemented like this and used in the app

File: src/models/test-types/test-type.model.ts [hunk 9]
```diff
@@ -0,0 +1,70 @@
+export interface TestTypeModel {
```
[the remark this answers is not in the document]
Comment (mgherceanu) [reply src/models/test-types/test-type.model.ts]: will conflict with other implementation.

[the remark this answers is not in the document]
Comment (adpetrescu) [reply src/models/test-types/test-type.model.ts]: This needs to be resolved. These interfaces are identical, so we shouldn't create one for each level.

[the remark this answers is not in the document]
Comment (mgherceanu) [reply src/models/test-types/test-type.model.ts]: done.

File: src/models/test-report.model.ts [hunk 10]
```diff
@@ -1,14 +1,17 @@
 import { VehicleModel } from './vehicle.model';
+import { PreparersModel } from "./preparers/preparers.model";
 
 export class TestReportModel {
     private createdAt: Date;
     private startTime: Date;
     private endTime: Date;
     private vehicles: VehicleModel[];
+    private preparer: PreparersModel | string;
```
Comment (adpetrescu) [inline src/models/test-report.model.ts]: I think that if the preparer is not present it should be null, not a hardcoded string. Can you please change this to be an optional and modify the logic in your methods accordingly?

**Pull request merged**